service/history/ndc/workflow_state_replicator.go

```diff
@@ -265,7 +265,29 @@ func (r *WorkflowStateReplicatorImpl) backfillHistory(
 	lastEventID int64,
 	lastEventVersion int64,
 	branchToken []byte,
-) (int64, error) {
+) (taskID int64, retError error) {
+
+	// Get the workflow lock to make sure no concurrent backfill history across multiple runs.
+	_, wfReleaseFn, err := r.workflowCache.GetOrCreateWorkflowExecution(
+		ctx,
+		r.shardContext,
+		namespaceID,
+		&commonpb.WorkflowExecution{
+			WorkflowId: workflowID,
+			RunId:      "",
```

Comment:
what is this doing? Is this to get current Run lock? Do we have a separate method for current run?

Reply:
Yes. I explicit use empty string to indicate this is to get the current run lock.